spdxexp/helpers.go

```diff
@@ -0,0 +1,24 @@
+package spdxexp
+
+// flatten will take an array of nested array and return
+// all nested elements in an array. e.g. [[1,2,[3]],4] -> [1,2,3,4]
+func flatten[T any](lists [][]T) []T {
+	var res []T
+	for _, list := range lists {
+		res = append(res, list...)
+	}
+	return res
+}
+
+// removeDuplicates will remove all duplicates from a slice
+func removeDuplicates[T string](sliceList []T) []T {
```

Comment:
I'm confused about `T string` here. Are there other string types, or could this just be a `removeDuplicateStrings` function?

Reply:
Whoops, it was a generic helper I grabbed from SO 🤦